include/wil/filesystem.h

```diff
@@ -1045,6 +1045,125 @@ namespace wil
     }
 #endif // _CPPUNWIND
 #endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)
+
+// Helpers to make the CreateFileW API easier to use. This segregates the OPEN_EXISTING "open an existing file"
+// cases from the "create a new file" that has 4 variations represented in the create_file_create_behavior enum.
+
+// https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew?devlangs=cpp&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(FILEAPI%252FCreateFileW)%3Bk(CreateFileW)%3Bk(DevLang-C%252B%252B)%3Bk(TargetOS-Windows)%26rd%3Dtrue

```

Comment:
Consider removing the `en-us/` part and cleaning up the stuff after the `?=` maybe.